packages/website/docs/creating-multi-source-autocompletes.md

```diff
@@ -2,3 +2,375 @@
+    return recentSearchesPlugin.data.getAlgoliaSearchParams({
+      hitsPerPage: 5,
+    });
+  },
+});
+const predefinedItems = [
+  {
+    label: 'Documentation',
+    url: 'https://autocomplete.algolia.com/',
+  },
+  {
+    label: 'GitHub',
+    url: 'https://github.com/algolia/autocomplete.js/tree/next',
+  },
+]
+const predefinedItemsPlugin = {
+  getSources() {
+    return [
+      {
+        getItems({ query }) {
+          return predefinedItems.filter(
+            (item) =>
+              !query || item.label.toLowerCase().includes(query.toLowerCase())
+          );
+        },
+        getItemUrl({ item }) {
+          return item.url;
+        },
+        templates: {
+          item({ item}) {
+              return (
+                <a href={item.url} className="aa-ItemContent aa-ItemLink aa-PredefinedItem">
+                  <div className="aa-ItemSourceIcon">
+                    <svg style="width:20px;height:20px" viewBox="0 0 24 24">
+                       <path fill="currentColor" d="M14,3V5H17.59L7.76,14.83L9.17,16.24L19,6.41V10H21V3M19,19H5V5H12V3H5C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V12H19V19Z" />
+                     </svg>
+                  </div>
+                  <div className="'aa-ItemTitle">{item.label}</div>
+                </a>
+              )
+        },
+        },
+      },
+    ];
+  },
+};
+
+Learn how to show different types of results in one autocomplete.
+
+:::note
+
+Following this tutorial requires an [Algolia](https://www.algolia.com/) application with the [Query Suggestions](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/query-suggestions/js/) feature enabled.
+
+:::
+
+While autocompletes with suggested searches is a ubiquitous search experience, rich multi-category autocompletes are becoming more and more popular.
+
+For example, if you search for something in your email inbox, your results could contain not just email threads, but also contacts, attachments, and more. Ecommerce stores often show suggested searches, products, blog posts, brands, and categories all in one autocomplete.
+
+Different results types are separated into different sections. This implicitly gives users a better understanding of what the items are, and what will happen if they select an item.
+
+The Autocomplete library lets you mix different item types in one autocomplete. To do so you need to return multiple [sources](sources#source) in the  [`getSources`](sources) option. This tutorial outlines how to combine [static predefined items](sources/#using-static-sources), [recent searches](using-recent-searches-plugin)] and [Query Suggestions](using-query-suggestions-plugin) in one autocomplete.
+
+Though it's not necessary, it uses plugins for each source. You can just as easily add different sources directly in [`getSources`](sources). However, it's recommend to encapsulate [source](sources) logic in a plugins since this makes it modular, reusable and sharable.
```

Comment:
Nice part 👍 

```suggestion
Though it's not necessary, it uses plugins for each source. You can just as easily add different sources directly in [`getSources`](sources). However, it's recommend to encapsulate [source](sources) logic in a plugin since this makes it modular, reusable and sharable.
```